### ISSUE TITLE
Deploy single get stuck indefinitely 

### ISSUE THREAD

**Issue by DBarthe** [datacharmer/dbdeployer#99]
Hello :)
Just tried dbdeployer and failed to deploy my first db instance.

**Describe the bug**
Downloading and unpacking was OK, but when running 'deploy', the command  stucks indefinitely with no messages.

**To Reproduce**
1. mkdir -p ~/opt/mysql
2. dbdeployer downloads get-by-version 5.7 --newest --dry-run
3. dbdeployer downloads get-by-version 5.7 --newest
4. dbdeployer unpack mysql-5.7.28-linux-glibc2.12-x86_64.tar.gz
5. dbdeployer  deploy single 5.7

**Expected behavior**
Deploy the database succesfuly or at least a message or what is being done or error...

I can't find any "verbose" or "debug" option...

**NO Screenshots**
```
$ dbdeployer   deploy single 5.7
# 5.7 => 5.7.28
[nothing here... the command still running, no disk IO, no CPU usage...] 
```

**Environment:**
 - OS: Ubuntu 18.04
 - dbdeployer version 1.42.0
 - tarball full name : see above
 - tarball origin : see above

**Hardware: (if applicable)**
quite OK...

**Additional context**
NOP

**Comment by datacharmer** [comment]
Hi
Thanks for reporting this issue.
Please look at these two files:
```
$HOME/sandboxes/msb_5_7_28/start.log
$HOME/sandboxes/msb_5_7_28/data/msandbox.err
```
If something bad is happening, it should be reported in one of the two files.

**Comment by DBarthe** [comment]
Thanks for your reactivity, folder sandboxes does not exist yet...

**Comment by datacharmer** [comment]
Do you have rights to create it? (it should have failed loudly if you don't, but we never know)

**Comment by DBarthe** [comment]
this is the output of strace (the last part only) :
```
...
read(3, 0xc000182600, 512)              = -1 EAGAIN (Resource temporarily unavailable)
futex(0xd34008, FUTEX_WAIT_PRIVATE, 0, NULL) = ? ERESTARTSYS (To be restarted if SA_RESTART is set)
--- SIGCHLD {si_signo=SIGCHLD, si_code=CLD_EXITED, si_pid=12610, si_uid=1000, si_status=0, si_utime=0, si_stime=0} ---
rt_sigreturn({mask=[]})                 = 202
futex(0xd34008, FUTEX_WAIT_PRIVATE, 0, NULL
```
Not sure what does it means, but seems to be related to a dead lock or something...

> Do you have rights to create it? (it should have failed loudly if you don't, but we never know)

Yes right is OK, i eventually created it manually.

**Comment by datacharmer** [comment]
Please try running the deployment command again with `--skip-library-check`.

**Comment by DBarthe** [comment]
Yep it worked, thanks :)

**Comment by DBarthe** [comment]
Everything is fine now, feel free to close the ticket, and thank you for providing this tool :+1:

**Comment by datacharmer** [comment]
Thanks for the report. Others have reported similar behavior, but I haven't been able to reproduce the failure. If I may, I'd like to ask you to run these commands (it's what dbdeployer calls without the option `--skip-library-check`):

```
ldconfig -p | grep numa
ldconfig -p | grep libaio
ldd ~/opt/mysql/5.7.28/bin/mysql
ldd ~/opt/mysql/5.7.28/bin/mysqld
```

**Comment by DBarthe** [comment]
Sure, see below :
```
$ ldconfig -p | grep numa
	libnuma.so.1 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libnuma.so.1
$ ldconfig -p | grep libaio
	libaio.so.1 (libc6,x86-64) => /lib/x86_64-linux-gnu/libaio.so.1
	libaio.so (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libaio.so
$ ldd ~/opt/mysql/5.7.28/bin/mysql
	linux-vdso.so.1 (0x00007fabdd492000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007fabdd04d000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007fabdce45000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007fabdcc41000)
	libncurses.so.5 => /lib/x86_64-linux-gnu/libncurses.so.5 (0x00007fabdca1e000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007fabdc695000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007fabdc2f7000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007fabdc0df000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007fabdbcee000)
	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007fabdbac4000)
	/lib64/ld-linux-x86-64.so.2 (0x00007fabdd26c000)
$ ldd ~/opt/mysql/5.7.28/bin/mysqld
	linux-vdso.so.1 (0x00007ffec4adf000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f888670b000)
	libaio.so.1 => /lib/x86_64-linux-gnu/libaio.so.1 (0x00007f8886509000)
	libnuma.so.1 => /usr/lib/x86_64-linux-gnu/libnuma.so.1 (0x00007f88862fe000)
	libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f88860c6000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f8885ec2000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f8885cba000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f8885931000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f8885593000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f888537b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f8884f8a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f888692a000)
```

**Comment by datacharmer** [comment]
I have made an experimental fix.
Would you try this [binary](https://file.io/MS4OKp) in your system to see if it fixes the problem? 
(Meaning running the deployment **without** --skip-library-check`)

**Comment by zarere** [comment]
@datacharmer I'm also using the same OS and having exactly the same issue with **Percona-XtraDB-Cluster-5.6.45-rel86.1-28.36.1.Linux.x86_64.ssl100.tar.gz**.
Trying with **--skip-library-check** is working ok but I can't try your version above since the link is dead already.

**Comment by datacharmer** [comment]
@zarere please try again. I have refreshed the binary

**Comment by zarere** [comment]
@datacharmer  Still error 404 when I click on the link

**Comment by datacharmer** [comment]
Darn. The expiration went wrong. This should be OK https://file.io/2tSmxS

**Comment by zarere** [comment]
@datacharmer This one  is also not working

**Comment by datacharmer** [comment]
Sorry about that. I should have made a release from the beginning. 
https://github.com/datacharmer/dbdeployer/releases/tag/1.43.0

**Comment by zarere** [comment]
@datacharmer  1.43 works fine without **--skip-library-check**

```
➜  ~ dbdeployer   deploy single 5.6.45                                                                    
Database installed in $HOME/sandboxes/msb_5_6_45
run 'dbdeployer usage single' for basic instructions'
. sandbox server started
➜  ~ dbdeployer  --version
dbdeployer version 1.43.0
```
Thanks a lot.

**Comment by datacharmer** [comment]
@zarere Thanks for testing. If you could offer some insight about the reason this is happening, it will help me  in the future testing. I made the fix blindly, as the issue doesn't happen in my servers, and I'd like to reproduce the problem.

**Comment by zarere** [comment]
@datacharmer If I need to be honest I have no idea for the reason.I just tested 1.42 today and saw that is not working, checked the reported issues and that's all.I will test something on my end and let you know.

**Comment by zarere** [comment]
@datacharmer Can this be because of the ssl versions that I have:

```
➜  ~ openssl version                                                                     
OpenSSL 1.1.1  11 Sep 2018
➜  ~ ls -lht /usr/lib/x86_64-linux-gnu |grep libssl                                             
-rw-r--r--  1 root root  321K Nov 26 15:53 libssl3.so
-rw-r--r--  1 root root  972K Nov 12 18:58 libssl.a
lrwxrwxrwx  1 root root    13 Nov 12 18:58 libssl.so -> libssl.so.1.1
-rw-r--r--  1 root root  564K Nov 12 18:58 libssl.so.1.1
-rw-r--r--  1 root root  417K Feb 26  2019 libssl.so.1.0.0

```
```
➜  ~ ldconfig -p | grep numa                                                
	libnuma.so.1 (libc6,x86-64) => /usr/lib/x86_64-linux-gnu/libnuma.so.1
```
```
➜  ~ ldconfig -p | grep libaio                                                                                                                                                                                           
	libaio.so.1 (libc6,x86-64) => /lib/x86_64-linux-gnu/libaio.so.1
```

```

➜  ~ ldd /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/opt/5.6.45/bin/mysql
	linux-vdso.so.1 (0x00007ffd815ca000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f30342b1000)
	libreadline.so.6 => /lib/x86_64-linux-gnu/libreadline.so.6 (0x00007f303406b000)
	libncurses.so.5 => /lib/x86_64-linux-gnu/libncurses.so.5 (0x00007f3033e48000)
	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f3033c1e000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f3033a01000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f30337f9000)
	libssl.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f3033591000)
	libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f303314e000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f3032f4a000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f3032bc1000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f3032823000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f303260b000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f303221a000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f30344d0000)
```
```
➜  ~ ldd /media/zarere/57f67066-a645-4244-8359-16c191fbadc1/opt/5.6.45/bin/mysqld
	linux-vdso.so.1 (0x00007ffe99ba7000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f242e0ae000)
	libaio.so.1 => /lib/x86_64-linux-gnu/libaio.so.1 (0x00007f242deac000)
	libnuma.so.1 => /usr/lib/x86_64-linux-gnu/libnuma.so.1 (0x00007f242dca1000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f242da84000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f242d87c000)
	libcrypt.so.1 => /lib/x86_64-linux-gnu/libcrypt.so.1 (0x00007f242d644000)
	libssl.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libssl.so.1.0.0 (0x00007f242d3dc000)
	libcrypto.so.1.0.0 => /usr/lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f242cf99000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f242cd95000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f242ca0c000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f242c66e000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f242c456000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f242c065000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f242e2cd000)
```

```
➜  ~            uname -mrs
Linux 5.0.0-29-generic x86_64
```
```
➜  ~ lsb_release -a                                                              
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.3 LTS
Release:	18.04
Codename:	bionic
```

**Comment by datacharmer** [comment]
@zarere Probably not. I have the same version, but it doesn't happen.
What kind of server are you using? Bare metal, virtual machine, container over bare metal, container over virtual machine?

**Comment by zarere** [comment]
Bare metal

**Comment by zarere** [comment]
@datacharmer So you are also having **libssl.so.1.1** and **libssl.so.1.0.0** at the same time?

**Comment by datacharmer** [comment]
Yes. The only difference I see with your setup is the kernel version

**Comment by zarere** [comment]
@datacharmer Well I can not understand why the issue was happening at my end then.

**Comment by zarere** [comment]
@datacharmer If you need any other tests let me know.

**Comment by dveeden** [comment]
I was hitting this bug when setting up a MySQL 8.0.18 sandbox on Fedora 31 with kernel 5.3.14 and ldconfig(glibc) version 2.30

**Comment by datacharmer** [comment]
@dveeden Does the temporary release below fix your problem?
https://github.com/datacharmer/dbdeployer/releases/tag/1.43.0

**Comment by dveeden** [comment]
> @dveeden Does the temporary release below fix your problem?
> https://github.com/datacharmer/dbdeployer/releases/tag/1.43.0
Yes looks like that works

**Comment by datacharmer** [comment]
Fixed in release [1.43.1](https://github.com/datacharmer/dbdeployer/releases/tag/1.43.1)
